presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -149,6 +155,16 @@ class ResponseHandler : public proxygen::HTTPTransactionHandler {
 
   void onBody(std::unique_ptr<folly::IOBuf> chain) noexcept override {
     if ((chain != nullptr) && !response_->hasError()) {
+      if (type_ == common::PRESTO_ANNOUNCER) {
+        REPORT_ADD_STAT_VALUE(kCounterHttpClientPrestoAnnouncerNumOfOnBody);
+        REPORT_ADD_HISTOGRAM_VALUE(
+            kCounterHttpClientPrestoAnnouncerSizeOfOnBodyChain,
+            chain->length());
+      } else if (type_ == common::PRESTO_EXCHANGE) {
+        REPORT_ADD_STAT_VALUE(kCounterHttpClientPrestoExchangeNumOfOnBody);
+        REPORT_ADD_HISTOGRAM_VALUE(
+            kCounterHttpClientPrestoExchangeSizeOfOnBodyChain, chain->length());
```

Comment:
Extract chain length out to reduce computation

Reply:
Can we just look servers request and response bytes counters? Announcement is based on a hard-coded message. Everything else is request/response worker to worker or worker to coordinator.

Reply:
You cannot just report exchange http client related counters inside the HttpClient class itself. You can do exchange related counters outside but you won't be able to get the body info because outside does not have access to onBody insights. 

Server request and response bytes counters are not the same as onBody buffers. I think I mentioned before that proxygen is chopping the response into smaller pieces and we need a better understanding on that because the piece of buffer we (prestissimo/velox) directly deal with is the onBody buffer.